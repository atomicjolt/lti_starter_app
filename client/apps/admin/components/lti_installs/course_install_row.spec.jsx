import React from 'react';
import TestRenderer from 'react-test-renderer';
import _ from 'lodash';
import CourseInstallRow from './course_install_row';

describe('lti installs course install row', () => {

  let result;
  let instance;
  const courseId = 123;
  const courseName = 'courseName';
  const installedToolId = 12;

  const applicationInstance = {
    name: 'application_name',
    lti_key: 'lti_key',
    lti_secret: 'lti_secret',
    lti_config_xml: 'lti_config_xml',
    site: {
      url: 'example.com'
    }
  };
  const installedTool = {
    id: installedToolId
  };
  const canvasRequest = () => {};

  beforeEach(() => {
    result = TestRenderer.create(
      <CourseInstallRow
        applicationInstance={applicationInstance}
        installedTool={installedTool}
        canvasRequest={canvasRequest}
        courseName={courseName}
        courseId={courseId}
      />);
    instance = result.root;
  });

  it('renders', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders row', () => {
    const a = instance.findByType('a');
    expect(a.props.href).toEqual('example.com/courses/123/external_tools/12');
  });

  it('renders buttons', () => {
    const installButton = instance.findByType('button');
    expect(installButton.props.children).toBe('Uninstall');
  });

  it('renders link without external tool', () => {
    result = TestRenderer.create(
      <CourseInstallRow
        applicationInstance={applicationInstance}
        canvasRequest={canvasRequest}
        courseName={courseName}
        courseId={courseId}
      />);
    const link = instance.findByType('a');
    expect(link).toBeDefined();
    const installLink = `example.com/courses/${courseId}`;
    const found = _.includes(link.props.href, installLink);
    expect(found).toBe(true);
  });

  it('renders link with external tool', () => {
    const link = instance.findByType('a');
    expect(link).toBeDefined();
    const installLink = `example.com/courses/${props.courseId}/external_tools/${installedToolId}`;
    const found = _.includes(link.props.href, installLink);
    expect(found).toBe(true);
  });

});
