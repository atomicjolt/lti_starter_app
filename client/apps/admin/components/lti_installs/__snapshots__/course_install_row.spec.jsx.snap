// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti installs course install row matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseInstallRow
    applicationInstance={
      Object {
        "lti_config_xml": "lti_config_xml",
        "lti_key": "lti_key",
        "lti_secret": "lti_secret",
        "name": "application_name",
        "site": Object {
          "url": "example.com",
        },
      }
    }
    canvasRequest={[Function]}
    courseId={123}
    courseName="courseName"
    installedTool={
      Object {
        "id": 12,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <div
            className="c-table--inactive"
          >
            <a
              href="example.com/courses/123/external_tools/12"
              rel="noopener noreferrer"
              target="_blank"
            >
              courseName
            </a>
          </div>
        </td>,
        <td>
          <button
            className="c-btn c-btn--gray"
            onClick={[Function]}
          >
            Uninstall
          </button>
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="c-table--inactive"
          >
            <a
              href="example.com/courses/123/external_tools/12"
              rel="noopener noreferrer"
              target="_blank"
            >
              courseName
            </a>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="example.com/courses/123/external_tools/12"
              rel="noopener noreferrer"
              target="_blank"
            >
              courseName
            </a>,
            "className": "c-table--inactive",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "courseName",
              "href": "example.com/courses/123/external_tools/12",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "courseName",
            "type": "a",
          },
          "type": "div",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="c-btn c-btn--gray"
            onClick={[Function]}
          >
            Uninstall
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Uninstall",
            "className": "c-btn c-btn--gray",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Uninstall",
          "type": "button",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <div
              className="c-table--inactive"
            >
              <a
                href="example.com/courses/123/external_tools/12"
                rel="noopener noreferrer"
                target="_blank"
              >
                courseName
              </a>
            </div>
          </td>,
          <td>
            <button
              className="c-btn c-btn--gray"
              onClick={[Function]}
            >
              Uninstall
            </button>
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="c-table--inactive"
            >
              <a
                href="example.com/courses/123/external_tools/12"
                rel="noopener noreferrer"
                target="_blank"
              >
                courseName
              </a>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="example.com/courses/123/external_tools/12"
                rel="noopener noreferrer"
                target="_blank"
              >
                courseName
              </a>,
              "className": "c-table--inactive",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "courseName",
                "href": "example.com/courses/123/external_tools/12",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "courseName",
              "type": "a",
            },
            "type": "div",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="c-btn c-btn--gray"
              onClick={[Function]}
            >
              Uninstall
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Uninstall",
              "className": "c-btn c-btn--gray",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Uninstall",
            "type": "button",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
